source/apiVolontaria/volunteer/tests/tests_model_Event.py

```diff
@@ -0,0 +1,245 @@
+from rest_framework.test import APIClient, APITransactionTestCase
+
+from django.db import IntegrityError
+from django.utils import timezone
+
+from apiVolontaria.factories import UserFactory, AdminFactory
+from location.models import Address, StateProvince, Country
+from ..models import Cell, TaskType, Cycle, Event
+
+
+class EventTests(APITransactionTestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+        self.user = UserFactory()
+        self.user.set_password('Test123!')
+        self.user.save()
+
+        self.admin = AdminFactory()
+        self.admin.set_password('Test123!')
+        self.admin.save()
+
+        self.random_country = Country.objects.create(
+            name="random country",
+            iso_code="RC",
+        )
+        self.random_state_province = StateProvince.objects.create(
+            name="random state",
+            iso_code="RS",
+            country=self.random_country,
+        )
+        self.address = Address.objects.create(
+            address_line1='random address 1',
+            postal_code='RAN DOM',
+            city='random city',
+            state_province=self.random_state_province,
+            country=self.random_country,
```

Comment:
I thought about this while creating the `Cell` model: should we use factories for `Address`, `Country`, `StateProvince`, `Cycle`, `tasktype`, `Cell` ?

(I am not asking you to do it, I am just asking so we can plan it for the future)

Reply:
Yeah! It will be a GREAT enhancement!

Reply:
New issue about this enhancement -> #53